### PR TITLE
fix some python3 issues, enable python3 travis

### PR DESCRIPTION
**Pull request by gijzelaerr**

### CODE REVIEW

File: cmake/FindNUMPY.cmake [hunk 1]
```diff
@@ -39,11 +39,7 @@
 #============================================================================
 
 # Finding NumPy involves calling the Python interpreter
-if(NumPy_FIND_REQUIRED)
-    find_package(PythonInterp REQUIRED)
-else()
-    find_package(PythonInterp)
-endif()
+find_package(PythonInterp REQUIRED)
```
Comment (tammojan) [inline cmake/FindNUMPY.cmake]: Why did you change this? If FindNumPy is now called without REQUIRED, it will fail if there is no interpreter, right?

Comment (gijzelaerr) [inline cmake/FindNUMPY.cmake]: I don't see how you can find a Numpy installation without a Python interpreter.

Comment (gijzelaerr) [inline cmake/FindNUMPY.cmake]: maybe you are right there, you don't want to fail when numpy is not a requirement.
